hack/install.sh

```diff
@@ -209,30 +229,51 @@ do_install() {
 				curl='curl -sSL'
 			fi
 			(
-				set -x
-				if [ "https://get.docker.com/" = "$url" ]; then
-					$sh_c "apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 36A1D7869245C8950F966E92D8576A8BA88D21E9"
-				elif [ "https://test.docker.com/" = "$url" ]; then
-					$sh_c "apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 740B314AE3941731B942C66ADF4FD13717AAD7D6"
-				elif [ "https://experimental.docker.com/" = "$url" ]; then
-					$sh_c "apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys E33FF7BF5C91D50A6F91FFFD4CC38D40F9A96B49"
-				else
-					$sh_c "$curl ${url}gpg | apt-key add -"
-				fi
-				$sh_c "mkdir -p /etc/apt/sources.list.d"
-				$sh_c "echo deb ${url}ubuntu docker main > /etc/apt/sources.list.d/docker.list"
-				$sh_c 'sleep 3; apt-get update; apt-get install -y -q lxc-docker'
+			set -x
+			$sh_c "apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D"
+			$sh_c "mkdir -p /etc/apt/sources.list.d"
+			$sh_c "echo deb https://apt.dockerproject.org/repo ${lsb_dist}-${dist_version} ${repo} > /etc/apt/sources.list.d/docker.list"
+			$sh_c 'sleep 3; apt-get update; apt-get install -y -q docker-engine'
 			)
 			if command_exists docker && [ -e /var/run/docker.sock ]; then
 				(
-					set -x
-					$sh_c 'docker version'
+				set -x
+				$sh_c 'docker version'
 				) || true
 			fi
 			echo_docker_as_nonroot
 			exit 0
 			;;
 
+		fedora|centos)
+			cat >/etc/yum.repos.d/docker${repo}.repo <<-EOF
+			[docker${repo}repo]
+			name=Docker ${repo} Repository
+			baseurl=https://yum.dockerproject.org/repo/${repo}/${lsb_dist}/${dist_version}
+			enabled=1
+			gpgcheck=1
+			gpgkey=https://yum.dockerproject.org/gpg
+			EOF
+			if [ "$lsb_dist" = "fedora" ] && [ "$dist_version" -ge "22" ]; then
+				(
+				set -x
+				$sh_c 'sleep 3; dnf -y -q install docker-engine'
+				)
+			else
+				(
+				set -x
+				$sh_c 'sleep 3; yum -y -q install docker-engine'
+				)
+			fi
+			if command_exists docker && [ -e /var/run/docker.sock ]; then
+				(
+				set -x
+				$sh_c 'docker version'
+				) || true
+			fi
```

Comment:
Since we're repeating this `docker version` block a couple times now, do you think we should maybe move it into `echo_docker_as_nonroot` instead?